src/anchoredinterval.jl

```diff
@@ -386,8 +386,8 @@ canonicalize(target_type::Type{P}, p::P) where P <: Period = p
 
 ##### TIME ZONES #####
 
-function TimeZones.astimezone(i::AnchoredInterval{P, ZonedDateTime, L, R}, tz::TimeZone) where {P,L,R}
+function TimeZones.astimezone(i::AnchoredInterval{P, T, L, R}, tz::TimeZone) where {P,T,L,R}
     return AnchoredInterval{P, ZonedDateTime, L, R}(astimezone(anchor(i), tz))
```

Comment:
```suggestion
    zdt = astimezone(anchor(i), tz)
    return AnchoredInterval{P, typeof(zdt), L, R}(zdt)
```
This could be done in the future. Probably this type of change would have to happen in a bunch of places for it to work and there's no use case right now, though, so we can hold off.

Reply:
Added an issue for it. Might want to do that at the same time that we try to make TimeZones an optional dependency.

https://github.com/invenia/Intervals.jl/issues/212